pacman/utilities/utility_calls.py

```diff
@@ -213,3 +214,66 @@ def get_key_ranges(key, mask):
         generated_key[remaining_zeros] = \
             expand_to_bit_array(value)[-len(remaining_zeros):]
         yield compress_from_bit_array(generated_key), n_keys
+
+
+def get_n_bits(n_values):
+    """ Determine how many bits are required for the given number of values
+
+    :param int n_values: the number of values (starting at 0)
+    :return: the number of bits required to express that many values
+    :rtype: int
+    """
+    if n_values == 0:
+        return 0
+    if n_values == 1:
+        return 1
+    return int(math.ceil(math.log(n_values, 2)))
```

Comment:
math.log2

Reply:
Note other existing places can be fixed too

Reply:
Fixed